flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStResourceContainer.java

```diff
@@ -0,0 +1,544 @@
+
+    private final boolean enableStatistics;
+
+    /** The handles to be closed when the container is closed. */
+    private final ArrayList<AutoCloseable> handlesToClose;
+
+    @Nullable private Path relocatedDbLogBaseDir;
+
+    @VisibleForTesting
+    public ForStResourceContainer() {
+        this(new Configuration(), null, null, null, null, false);
+    }
+
+    @VisibleForTesting
+    public ForStResourceContainer(@Nullable ForStOptionsFactory optionsFactory) {
+        this(new Configuration(), optionsFactory, null, null, null, false);
+    }
+
+    @VisibleForTesting
+    public ForStResourceContainer(
+            @Nullable ForStOptionsFactory optionsFactory,
+            @Nullable OpaqueMemoryResource<ForStSharedResources> sharedResources) {
+        this(new Configuration(), optionsFactory, sharedResources, null, null, false);
+    }
+
+    public ForStResourceContainer(
+            ReadableConfig configuration,
+            @Nullable ForStOptionsFactory optionsFactory,
+            @Nullable OpaqueMemoryResource<ForStSharedResources> sharedResources,
+            @Nullable File localBasePath,
+            @Nullable URI remoteBasePath,
+            boolean enableStatistics) {
+
+        this.configuration = configuration;
+        this.optionsFactory = optionsFactory;
+        this.sharedResources = sharedResources;
+
+        this.localBasePath = localBasePath;
+        this.localForStPath = localBasePath != null ? new File(localBasePath, DB_DIR_STRING) : null;
+        this.remoteBasePath = remoteBasePath;
+        this.remoteForStPath =
+                remoteBasePath != null ? remoteBasePath.resolve(DB_DIR_STRING) : null;
+
+        this.enableStatistics = enableStatistics;
+        this.handlesToClose = new ArrayList<>();
+    }
+
+    /** Gets the ForSt {@link DBOptions} to be used for ForSt instances. */
+    public DBOptions getDbOptions() {
+        // initial options from common profile
+        DBOptions opt = createBaseCommonDBOptions();
+        handlesToClose.add(opt);
+
+        // load configurable options on top of pre-defined profile
+        setDBOptionsFromConfigurableOptions(opt);
+
+        // add user-defined options factory, if specified
+        if (optionsFactory != null) {
+            opt = optionsFactory.createDBOptions(opt, handlesToClose);
+        }
+
+        // add necessary default options
+        opt = opt.setCreateIfMissing(true).setAvoidFlushDuringShutdown(true);
+
+        // if sharedResources is non-null, use the write buffer manager from it.
+        if (sharedResources != null) {
+            opt.setWriteBufferManager(sharedResources.getResourceHandle().getWriteBufferManager());
+        }
+
+        if (enableStatistics) {
+            Statistics statistics = new Statistics();
+            opt.setStatistics(statistics);
+            handlesToClose.add(statistics);
+        }
+
+        if (remoteForStPath != null) {
```

Comment:
What if the `remoteForStPath` is null?

Reply:
As described in its config, it will rollback to local dir and works just as RocksDB.